### PR TITLE
php.packages.grumphp: init at 1.8.1

### PR DESCRIPTION
**Pull request by drupol**
###### Motivation for this change

Add the [Grumphp](https://github.com/phpro/grumphp) package in NixOS.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by etu** [comment]
The commit should reference the attribute: php.packages.grumphp

File: pkgs/development/php-packages/grumphp/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    url = "https://github.com/phpro/${pname}/releases/download/v${version}/${pname}.phar";
+    sha256 = "sha256-3XPMyH2F3ZfRr8DmvlBY3Z6uolhaRraQxwKIskIwPq8=";
+  };
+
+  dontUnpack = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    mkdir -p $out/bin
```
Comment (jtojnar) [inline pkgs/development/php-packages/grumphp/default.nix:16]: When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`. See <https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md> for more info.

Comment (drupol) [reply pkgs/development/php-packages/grumphp/default.nix:18]: Right ! I took inspiration from already existing packages. Should we also reproduce this on all other packages overriding `installPhase` too ?

**Comment by drupol** [comment]
Dear @etu, @globin, @Ma27, @talyz,

With this PR, I'm trying to introduce [Grumphp](https://github.com/phpro/grumphp) in Nix.

Are you ok to set the maintainers to `teams.php.members` ?

To some extend, should this be the default for all the new PHP related packages yet to come ?

**Comment by etu** [comment]
@drupol We usually tag the team for PHP packages and if there's a person who's not part of the team who's willing to maintain it we add that person as well as the team to that package :)

File: pkgs/development/php-packages/grumphp/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    makeWrapper ${php}/bin/php $out/bin/grumphp \
+      --add-flags "$out/libexec/${pname}/grumphp.phar"
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A PHP code-quality tool";
+    homepage = "https://github.com/phpro/grumphp";
+    license = licenses.mit;
+    maintainers = teams.php.members;
```
Comment (etu) [inline pkgs/development/php-packages/grumphp/default.nix:28]: I've built this on every version of PHP and executed it in all of them.

It works in PHP 8.0 and PHP 8.1. But on PHP 7.4 it doesn't work and tells me that it requires PHP 8.0 or newer.

```suggestion
    maintainers = teams.php.members;
    broken = versionOlder php.version "8.0";
```

Comment (drupol) [reply pkgs/development/php-packages/grumphp/default.nix:29]: Done!

Comment (drupol) [reply pkgs/development/php-packages/grumphp/default.nix:29]: I also fixed the conflicting issue with `master` branch.

**Comment by drupol** [comment]
Thanks !

**Pull request merged**